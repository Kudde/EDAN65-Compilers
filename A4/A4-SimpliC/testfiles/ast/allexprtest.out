Program
  List
    Fun
      IntType ID="int"
      IdDecl ID="expr"
      Param
        List
      Block
        List
          DeclStmt
            IntType ID="int"
            IdDecl ID="x"
            Opt
          DeclStmt
            IntType ID="int"
            IdDecl ID="y"
            Opt
          DeclStmt
            IntType ID="int"
            IdDecl ID="k"
            Opt
          DeclStmt
            IntType ID="int"
            IdDecl ID="ans"
            Opt
          AssignStmt
            IdUse ID="x"
            Literal NUMERAL="2"
          AssignStmt
            IdUse ID="y"
            Literal NUMERAL="3"
          AssignStmt
            IdUse ID="k"
            Literal NUMERAL="5"
          AssignStmt
            IdUse ID="ans"
            Mod
              Sub
                Plus
                  IdUse ID="x"
                  Mul
                    Plus
                      Literal NUMERAL="2"
                      Literal NUMERAL="3"
                    Literal NUMERAL="2"
                Div
                  Literal NUMERAL="4"
                  Literal NUMERAL="2"
              Literal NUMERAL="2"
