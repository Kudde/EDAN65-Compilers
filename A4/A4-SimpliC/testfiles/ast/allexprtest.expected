Program
  List
    Fun
      TypeDecl Type="int"
      IdDecl ID="expr"
      Param
        List
      Block
        List
          DeclStmt
            TypeDecl Type="int"
            IdDecl ID="x"
            Opt
          DeclStmt
            TypeDecl Type="int"
            IdDecl ID="y"
            Opt
          DeclStmt
            TypeDecl Type="int"
            IdDecl ID="k"
            Opt
          DeclStmt
            TypeDecl Type="int"
            IdDecl ID="ans"
            Opt
          AssignStmt
            IdUse ID="x"
            Literal NUMERAL="2"
          AssignStmt
            IdUse ID="y"
            Literal NUMERAL="3"
          AssignStmt
            IdUse ID="k"
            Literal NUMERAL="5"
          AssignStmt
            IdUse ID="ans"
            Mod
              Sub
                Plus
                  IdUse ID="x"
                  Mul
                    Plus
                      Literal NUMERAL="2"
                      Literal NUMERAL="3"
                    Literal NUMERAL="2"
                Div
                  Literal NUMERAL="4"
                  Literal NUMERAL="2"
              Literal NUMERAL="2"
